DatePicker: showTime not detected automatically for LocalDateTime

In the migration document (8->9), it is stated that the attribute showTime is detected automatically based on value-binding to LocalDateTime. I cannot nor verify this for the latest PF9-snapshot. Could somebody please check and fix this. Many thanks in advance.

unfortunately not. build fails. 

```
/primefaces$ mvn clean jetty:run
[INFO] Scanning for projects...
Downloading from apache-snapshots: https://repository.apache.org/snapshots/org/apache/maven/plugins/maven-antrun-plugin/1.3/maven-antrun-plugin-1.3.pom
Downloading from central: https://repo.maven.apache.org/maven2/org/apache/maven/plugins/maven-antrun-plugin/1.3/maven-antrun-plugin-1.3.pom
Downloaded from central: https://repo.maven.apache.org/maven2/org/apache/maven/plugins/maven-antrun-plugin/1.3/maven-antrun-plugin-1.3.pom (4.7 kB at 67 kB/s)
Downloading from apache-snapshots: https://repository.apache.org/snapshots/org/apache/maven/plugins/maven-plugins/12/maven-plugins-12.pom
Downloading from central: https://repo.maven.apache.org/maven2/org/apache/maven/plugins/maven-plugins/12/maven-plugins-12.pom
Downloaded from central: https://repo.maven.apache.org/maven2/org/apache/maven/plugins/maven-plugins/12/maven-plugins-12.pom (12 kB at 559 kB/s)
Downloading from apache-snapshots: https://repository.apache.org/snapshots/org/apache/maven/maven-parent/9/maven-parent-9.pom
Downloading from central: https://repo.maven.apache.org/maven2/org/apache/maven/maven-parent/9/maven-parent-9.pom
Downloaded from central: https://repo.maven.apache.org/maven2/org/apache/maven/maven-parent/9/maven-parent-9.pom (33 kB at 1.1 MB/s)
Downloading from apache-snapshots: https://repository.apache.org/snapshots/org/apache/maven/plugins/maven-antrun-plugin/1.3/maven-antrun-plugin-1.3.jar
Downloading from central: https://repo.maven.apache.org/maven2/org/apache/maven/plugins/maven-antrun-plugin/1.3/maven-antrun-plugin-1.3.jar
Downloaded from central: https://repo.maven.apache.org/maven2/org/apache/maven/plugins/maven-antrun-plugin/1.3/maven-antrun-plugin-1.3.jar (24 kB at 1.1 MB/s)
Downloading from apache-snapshots: https://repository.apache.org/snapshots/org/apache/maven/plugins/maven-assembly-plugin/2.2-beta-5/maven-assembly-plugin-2.2-beta-5.pom
Downloading from central: https://repo.maven.apache.org/maven2/org/apache/maven/plugins/maven-assembly-plugin/2.2-beta-5/maven-assembly-plugin-2.2-beta-5.pom
Downloaded from central: https://repo.maven.apache.org/maven2/org/apache/maven/plugins/maven-assembly-plugin/2.2-beta-5/maven-assembly-plugin-2.2-beta-5.pom (15 kB at 758 kB/s)
Downloading from apache-snapshots: https://repository.apache.org/snapshots/org/apache/maven/plugins/maven-plugins/16/maven-plugins-16.pom
Downloading from central: https://repo.maven.apache.org/maven2/org/apache/maven/plugins/maven-plugins/16/maven-plugins-16.pom
Downloaded from central: https://repo.maven.apache.org/maven2/org/apache/maven/plugins/maven-plugins/16/maven-plugins-16.pom (13 kB at 647 kB/s)
Downloading from apache-snapshots: https://repository.apache.org/snapshots/org/apache/maven/plugins/maven-assembly-plugin/2.2-beta-5/maven-assembly-plugin-2.2-beta-5.jar
Downloading from central: https://repo.maven.apache.org/maven2/org/apache/maven/plugins/maven-assembly-plugin/2.2-beta-5/maven-assembly-plugin-2.2-beta-5.jar
Downloaded from central: https://repo.maven.apache.org/maven2/org/apache/maven/plugins/maven-assembly-plugin/2.2-beta-5/maven-assembly-plugin-2.2-beta-5.jar (209 kB at 5.2 MB/s)
Downloading from apache-snapshots: https://repository.apache.org/snapshots/org/apache/maven/plugins/maven-dependency-plugin/2.8/maven-dependency-plugin-2.8.pom
Downloading from central: https://repo.maven.apache.org/maven2/org/apache/maven/plugins/maven-dependency-plugin/2.8/maven-dependency-plugin-2.8.pom
Downloaded from central: https://repo.maven.apache.org/maven2/org/apache/maven/plugins/maven-dependency-plugin/2.8/maven-dependency-plugin-2.8.pom (11 kB at 573 kB/s)
Downloading from apache-snapshots: https://repository.apache.org/snapshots/org/apache/maven/plugins/maven-dependency-plugin/2.8/maven-dependency-plugin-2.8.jar
Downloading from central: https://repo.maven.apache.org/maven2/org/apache/maven/plugins/maven-dependency-plugin/2.8/maven-dependency-plugin-2.8.jar
Downloaded from central: https://repo.maven.apache.org/maven2/org/apache/maven/plugins/maven-dependency-plugin/2.8/maven-dependency-plugin-2.8.jar (153 kB at 5.3 MB/s)
Downloading from apache-snapshots: https://repository.apache.org/snapshots/org/apache/maven/plugins/maven-release-plugin/2.1/maven-release-plugin-2.1.pom
Downloading from central: https://repo.maven.apache.org/maven2/org/apache/maven/plugins/maven-release-plugin/2.1/maven-release-plugin-2.1.pom
Downloaded from central: https://repo.maven.apache.org/maven2/org/apache/maven/plugins/maven-release-plugin/2.1/maven-release-plugin-2.1.pom (8.2 kB at 411 kB/s)
Downloading from apache-snapshots: https://repository.apache.org/snapshots/org/apache/maven/release/maven-release/2.1/maven-release-2.1.pom
Downloading from central: https://repo.maven.apache.org/maven2/org/apache/maven/release/maven-release/2.1/maven-release-2.1.pom
Downloaded from central: https://repo.maven.apache.org/maven2/org/apache/maven/release/maven-release/2.1/maven-release-2.1.pom (7.1 kB at 307 kB/s)
Downloading from apache-snapshots: https://repository.apache.org/snapshots/org/apache/maven/plugins/maven-release-plugin/2.1/maven-release-plugin-2.1.jar
Downloading from central: https://repo.maven.apache.org/maven2/org/apache/maven/plugins/maven-release-plugin/2.1/maven-release-plugin-2.1.jar
Downloaded from central: https://repo.maven.apache.org/maven2/org/apache/maven/plugins/maven-release-plugin/2.1/maven-release-plugin-2.1.jar (40 kB at 1.8 MB/s)
Downloading from apache-snapshots: https://repository.apache.org/snapshots/com/github/eirslett/frontend-maven-plugin/1.9.1/frontend-maven-plugin-1.9.1.pom
Downloading from central: https://repo.maven.apache.org/maven2/com/github/eirslett/frontend-maven-plugin/1.9.1/frontend-maven-plugin-1.9.1.pom
Downloaded from central: https://repo.maven.apache.org/maven2/com/github/eirslett/frontend-maven-plugin/1.9.1/frontend-maven-plugin-1.9.1.pom (4.9 kB at 222 kB/s)
Downloading from apache-snapshots: https://repository.apache.org/snapshots/com/github/eirslett/frontend-plugins/1.9.1/frontend-plugins-1.9.1.pom
Downloading from central: https://repo.maven.apache.org/maven2/com/github/eirslett/frontend-plugins/1.9.1/frontend-plugins-1.9.1.pom
Downloaded from central: https://repo.maven.apache.org/maven2/com/github/eirslett/frontend-plugins/1.9.1/frontend-plugins-1.9.1.pom (6.6 kB at 314 kB/s)
Downloading from apache-snapshots: https://repository.apache.org/snapshots/com/github/eirslett/frontend-maven-plugin/1.9.1/frontend-maven-plugin-1.9.1.jar
Downloading from central: https://repo.maven.apache.org/maven2/com/github/eirslett/frontend-maven-plugin/1.9.1/frontend-maven-plugin-1.9.1.jar
Downloaded from central: https://repo.maven.apache.org/maven2/com/github/eirslett/frontend-maven-plugin/1.9.1/frontend-maven-plugin-1.9.1.jar (35 kB at 1.7 MB/s)
Downloading from apache-snapshots: https://repository.apache.org/snapshots/org/codehaus/mojo/maven-metadata.xml
Downloading from central: https://repo.maven.apache.org/maven2/org/codehaus/mojo/maven-metadata.xml
Downloading from central: https://repo.maven.apache.org/maven2/org/apache/maven/plugins/maven-metadata.xml
Downloading from apache-snapshots: https://repository.apache.org/snapshots/org/apache/maven/plugins/maven-metadata.xml
Downloaded from central: https://repo.maven.apache.org/maven2/org/apache/maven/plugins/maven-metadata.xml (14 kB at 688 kB/s)
Downloaded from central: https://repo.maven.apache.org/maven2/org/codehaus/mojo/maven-metadata.xml (20 kB at 436 kB/s)
Downloaded from apache-snapshots: https://repository.apache.org/snapshots/org/apache/maven/plugins/maven-metadata.xml (8.5 kB at 110 kB/s)
[INFO] ------------------------------------------------------------------------
[INFO] BUILD FAILURE
[INFO] ------------------------------------------------------------------------
[INFO] Total time:  2.178 s
[INFO] Finished at: 2020-10-28T13:13:05+01:00
[INFO] ------------------------------------------------------------------------
[ERROR] No plugin found for prefix 'jetty' in the current project and in the plugin groups [org.apache.maven.plugins, org.codehaus.mojo] available from the repositories [local (/srv/bfabric/.m2/repository), apache-snapshots (https://repository.apache.org/snapshots/), central (https://repo.maven.apache.org/maven2)] -> [Help 1]
[ERROR] 
[ERROR] To see the full stack trace of the errors, re-run Maven with the -e switch.
[ERROR] Re-run Maven using the -X switch to enable full debug logging.
[ERROR] 
[ERROR] For more information about the errors and possible solutions, please read the following articles:
[ERROR] [Help 1] http://cwiki.apache.org/confluence/display/MAVEN/NoPluginFoundForPrefixException

```

i have now unzipped the file and build it successfully after having also checked out your latest fix.

however:  http://localhost:8080/primefaces-test/test.xhtml delivers

```
HTTP ERROR 500 javax.servlet.ServletException: javax.servlet.ServletException
URI:	/primefaces-test/test.xhtml
STATUS:	500
MESSAGE:	javax.servlet.ServletException: javax.servlet.ServletException
SERVLET:	Faces Servlet
CAUSED BY:	javax.servlet.ServletException: javax.servlet.ServletException
CAUSED BY:	javax.servlet.ServletException
CAUSED BY:	java.lang.NullPointerException
Caused by:
javax.servlet.ServletException: javax.servlet.ServletException
	at org.eclipse.jetty.server.handler.HandlerCollection.handle(HandlerCollection.java:162)
	at org.eclipse.jetty.server.handler.HandlerWrapper.handle(HandlerWrapper.java:127)
	at org.eclipse.jetty.server.Server.handle(Server.java:516)
	at org.eclipse.jetty.server.HttpChannel.lambda$handle$1(HttpChannel.java:383)
	at org.eclipse.jetty.server.HttpChannel.dispatch(HttpChannel.java:556)
	at org.eclipse.jetty.server.HttpChannel.handle(HttpChannel.java:375)
	at org.eclipse.jetty.server.HttpConnection.onFillable(HttpConnection.java:273)
	at org.eclipse.jetty.io.AbstractConnection$ReadCallback.succeeded(AbstractConnection.java:311)
	at org.eclipse.jetty.io.FillInterest.fillable(FillInterest.java:105)
	at org.eclipse.jetty.io.ChannelEndPoint$1.run(ChannelEndPoint.java:104)
	at org.eclipse.jetty.util.thread.QueuedThreadPool.runJob(QueuedThreadPool.java:773)
	at org.eclipse.jetty.util.thread.QueuedThreadPool$Runner.run(QueuedThreadPool.java:905)
	at java.lang.Thread.run(Thread.java:748)
Caused by: javax.servlet.ServletException
	at javax.faces.webapp.FacesServlet.service(FacesServlet.java:671)
	at org.eclipse.jetty.servlet.ServletHolder$NotAsync.service(ServletHolder.java:1418)
	at org.eclipse.jetty.servlet.ServletHolder.handle(ServletHolder.java:763)
	at org.eclipse.jetty.servlet.ServletHandler$CachedChain.doFilter(ServletHandler.java:1655)
	at org.eclipse.jetty.websocket.server.WebSocketUpgradeFilter.doFilter(WebSocketUpgradeFilter.java:228)
	at org.eclipse.jetty.servlet.FilterHolder.doFilter(FilterHolder.java:193)
	at org.eclipse.jetty.servlet.ServletHandler$CachedChain.doFilter(ServletHandler.java:1642)
	at org.eclipse.jetty.servlet.ServletHandler.doHandle(ServletHandler.java:572)
	at org.eclipse.jetty.server.handler.ScopedHandler.handle(ScopedHandler.java:143)
	at org.eclipse.jetty.security.SecurityHandler.handle(SecurityHandler.java:602)
	at org.eclipse.jetty.server.handler.HandlerWrapper.handle(HandlerWrapper.java:127)
	at org.eclipse.jetty.server.handler.ScopedHandler.nextHandle(ScopedHandler.java:235)
	at org.eclipse.jetty.server.session.SessionHandler.doHandle(SessionHandler.java:1612)
	at org.eclipse.jetty.server.handler.ScopedHandler.nextHandle(ScopedHandler.java:233)
	at org.eclipse.jetty.server.handler.ContextHandler.doHandle(ContextHandler.java:1434)
	at org.eclipse.jetty.server.handler.ScopedHandler.nextScope(ScopedHandler.java:188)
	at org.eclipse.jetty.servlet.ServletHandler.doScope(ServletHandler.java:512)
	at org.eclipse.jetty.server.session.SessionHandler.doScope(SessionHandler.java:1582)
	at org.eclipse.jetty.server.handler.ScopedHandler.nextScope(ScopedHandler.java:186)
	at org.eclipse.jetty.server.handler.ContextHandler.doScope(ContextHandler.java:1349)
	at org.eclipse.jetty.server.handler.ScopedHandler.handle(ScopedHandler.java:141)
	at org.eclipse.jetty.server.handler.ContextHandlerCollection.handle(ContextHandlerCollection.java:191)
	at org.eclipse.jetty.server.handler.HandlerCollection.handle(HandlerCollection.java:146)
	... 12 more
Caused by: java.lang.NullPointerException
	at org.primefaces.component.datepicker.DatePickerRenderer.encodeMarkup(DatePickerRenderer.java:124)
	at org.primefaces.component.calendar.BaseCalendarRenderer.encodeEnd(BaseCalendarRenderer.java:89)
	at org.primefaces.component.datepicker.DatePickerRenderer.encodeEnd(DatePickerRenderer.java:63)
	at javax.faces.component.UIComponentBase.encodeEnd(UIComponentBase.java:920)
	at javax.faces.component.UIComponent.encodeAll(UIComponent.java:1863)
	at javax.faces.render.Renderer.encodeChildren(Renderer.java:176)
	at javax.faces.component.UIComponentBase.encodeChildren(UIComponentBase.java:890)
	at javax.faces.component.UIComponent.encodeAll(UIComponent.java:1856)
	at javax.faces.component.UIComponent.encodeAll(UIComponent.java:1859)
	at javax.faces.component.UIComponent.encodeAll(UIComponent.java:1859)
	at com.sun.faces.application.view.FaceletViewHandlingStrategy.renderView(FaceletViewHandlingStrategy.java:456)
	at com.sun.faces.application.view.MultiViewHandler.renderView(MultiViewHandler.java:134)
	at javax.faces.application.ViewHandlerWrapper.renderView(ViewHandlerWrapper.java:337)
	at javax.faces.application.ViewHandlerWrapper.renderView(ViewHandlerWrapper.java:337)
	at com.sun.faces.lifecycle.RenderResponsePhase.execute(RenderResponsePhase.java:120)
	at com.sun.faces.lifecycle.Phase.doPhase(Phase.java:101)
	at com.sun.faces.lifecycle.LifecycleImpl.render(LifecycleImpl.java:219)
	at javax.faces.webapp.FacesServlet.service(FacesServlet.java:659)
```

--- a/src/main/java/org/primefaces/component/datepicker/DatePickerRenderer.java
+++ b/src/main/java/org/primefaces/component/datepicker/DatePickerRenderer.java
@@ -54,12 +54,38 @@ public class DatePickerRenderer extends BaseCalendarRenderer {
 
     @Override
     public void encodeEnd(FacesContext context, UIComponent component) throws IOException {
-        DatePicker datePicker = (DatePicker) component;
+        DatePicker datepicker = (DatePicker) component;
 
-        if (datePicker.getModel() instanceof LazyDateMetadataModel && ComponentUtils.isRequestSource(datePicker, context, "viewChange")) {
-            encodeDateMetadata(context, datePicker);
+        if (datepicker.getModel() instanceof LazyDateMetadataModel && ComponentUtils.isRequestSource(datepicker, context, "viewChange")) {
+            encodeDateMetadata(context, datepicker);
             return;
         }
+
+        String pattern = datepicker.getPattern() == null ? datepicker.calculatePattern() : datepicker.getPattern();
+
+        if (datepicker.isShowTimeWithoutDefault() == null) {
+            Class<?> type = datepicker.getTypeFromValueByValueExpression(context);
+
+            if (type != null) {
+                datepicker.setShowTime(LocalDateTime.class.isAssignableFrom(type));
+            }
+            else {
+                datepicker.setShowTime(CalendarUtils.hasTime(pattern));
+            }
+        }
+
+        if (datepicker.isTimeOnlyWithoutDefault() == null) {
+            Class<?> type = datepicker.getTypeFromValueByValueExpression(context);
+
+            if (type != null) {
+                datepicker.setTimeOnly(LocalTime.class.isAssignableFrom(type));
+            }
+        }
+
+        if (datepicker.isShowSecondsWithoutDefault() == null) {
+            datepicker.setShowSeconds(pattern.contains("s"));
+        }
+
         super.encodeEnd(context, component);
     }
 
@@ -99,31 +125,6 @@ public class DatePickerRenderer extends BaseCalendarRenderer {
     @Override
     protected void encodeMarkup(FacesContext context, UICalendar uicalendar, String value) throws IOException {
         DatePicker datepicker = (DatePicker) uicalendar;
-        String pattern = datepicker.getPattern() == null ? datepicker.calculatePattern() : datepicker.getPattern();
-
-        if (datepicker.isShowTimeWithoutDefault() == null) {
-            Class<?> type = datepicker.getTypeFromValueByValueExpression(context);
-
-            if (type != null) {
-                datepicker.setShowTime(LocalDateTime.class.isAssignableFrom(type));
-            }
-            else {
-                datepicker.setShowTime(CalendarUtils.hasTime(pattern));
-            }
-        }
-
-        if (datepicker.isTimeOnlyWithoutDefault() == null) {
-            Class<?> type = datepicker.getTypeFromValueByValueExpression(context);
-
-            if (type != null) {
-                datepicker.setTimeOnly(LocalTime.class.isAssignableFrom(type));
-            }
-        }
-
-        if (datepicker.isShowSecondsWithoutDefault() == null) {
-            datepicker.setShowSeconds(pattern.contains("s"));
-        }
-
         ResponseWriter writer = context.getResponseWriter();
         String clientId = datepicker.getClientId(context);
         String styleClass = datepicker.getStyleClass();

--- a/src/main/java/org/primefaces/component/datepicker/DatePickerRenderer.java
+++ b/src/main/java/org/primefaces/component/datepicker/DatePickerRenderer.java
@@ -54,36 +54,36 @@ public class DatePickerRenderer extends BaseCalendarRenderer {
 
     @Override
     public void encodeEnd(FacesContext context, UIComponent component) throws IOException {
-        DatePicker datepicker = (DatePicker) component;
+        DatePicker datePicker = (DatePicker) component;
 
-        if (datepicker.getModel() instanceof LazyDateMetadataModel && ComponentUtils.isRequestSource(datepicker, context, "viewChange")) {
-            encodeDateMetadata(context, datepicker);
+        if (datePicker.getModel() instanceof LazyDateMetadataModel && ComponentUtils.isRequestSource(datePicker, context, "viewChange")) {
+            encodeDateMetadata(context, datePicker);
             return;
         }
 
-        String pattern = datepicker.getPattern() == null ? datepicker.calculatePattern() : datepicker.getPattern();
+        String pattern = datePicker.getPattern() == null ? datePicker.calculatePattern() : datePicker.getPattern();
 
-        if (datepicker.isShowTimeWithoutDefault() == null) {
-            Class<?> type = datepicker.getTypeFromValueByValueExpression(context);
+        if (datePicker.isShowTimeWithoutDefault() == null) {
+            Class<?> type = datePicker.getTypeFromValueByValueExpression(context);
 
             if (type != null) {
-                datepicker.setShowTime(LocalDateTime.class.isAssignableFrom(type));
+                datePicker.setShowTime(LocalDateTime.class.isAssignableFrom(type));
             }
             else {
-                datepicker.setShowTime(CalendarUtils.hasTime(pattern));
+                datePicker.setShowTime(CalendarUtils.hasTime(pattern));
             }
         }
 
-        if (datepicker.isTimeOnlyWithoutDefault() == null) {
-            Class<?> type = datepicker.getTypeFromValueByValueExpression(context);
+        if (datePicker.isTimeOnlyWithoutDefault() == null) {
+            Class<?> type = datePicker.getTypeFromValueByValueExpression(context);
 
             if (type != null) {
-                datepicker.setTimeOnly(LocalTime.class.isAssignableFrom(type));
+                datePicker.setTimeOnly(LocalTime.class.isAssignableFrom(type));
             }
         }
 
-        if (datepicker.isShowSecondsWithoutDefault() == null) {
-            datepicker.setShowSeconds(pattern.contains("s"));
+        if (datePicker.isShowSecondsWithoutDefault() == null) {
+            datePicker.setShowSeconds(pattern.contains("s"));
         }
 
         super.encodeEnd(context, component);
@@ -124,25 +124,25 @@ public class DatePickerRenderer extends BaseCalendarRenderer {
 
     @Override
     protected void encodeMarkup(FacesContext context, UICalendar uicalendar, String value) throws IOException {
-        DatePicker datepicker = (DatePicker) uicalendar;
+        DatePicker datePicker = (DatePicker) uicalendar;
         ResponseWriter writer = context.getResponseWriter();
-        String clientId = datepicker.getClientId(context);
-        String styleClass = datepicker.getStyleClass();
+        String clientId = datePicker.getClientId(context);
+        String styleClass = datePicker.getStyleClass();
         styleClass = (styleClass == null) ? UICalendar.CONTAINER_CLASS : UICalendar.CONTAINER_CLASS + " " + styleClass;
         styleClass = DatePicker.CONTAINER_EXTENSION_CLASS + " " + styleClass;
         String inputId = clientId + "_input";
-        boolean inline = datepicker.isInline();
+        boolean inline = datePicker.isInline();
 
-        writer.startElement("span", datepicker);
+        writer.startElement("span", datePicker);
         writer.writeAttribute("id", clientId, null);
         writer.writeAttribute("class", styleClass, null);
 
-        if (datepicker.getStyle() != null) {
-            writer.writeAttribute("style", datepicker.getStyle(), null);
+        if (datePicker.getStyle() != null) {
+            writer.writeAttribute("style", datePicker.getStyle(), null);
         }
 
         //input
-        encodeInput(context, datepicker, inputId, value, !inline);
+        encodeInput(context, datePicker, inputId, value, !inline);
 
         writer.endElement("span");
 
@@ -150,132 +150,132 @@ public class DatePickerRenderer extends BaseCalendarRenderer {
 
     @Override
     protected void encodeScript(FacesContext context, UICalendar uicalendar, String value) throws IOException {
-        DatePicker datepicker = (DatePicker) uicalendar;
-        String clientId = datepicker.getClientId(context);
-        Locale locale = datepicker.calculateLocale(context);
-        String pattern = datepicker.calculateWidgetPattern();
+        DatePicker datePicker = (DatePicker) uicalendar;
+        String clientId = datePicker.getClientId(context);
+        Locale locale = datePicker.calculateLocale(context);
+        String pattern = datePicker.calculateWidgetPattern();
         WidgetBuilder wb = getWidgetBuilder(context);
-        wb.init("DatePicker", datepicker.resolveWidgetVar(context), clientId);
+        wb.init("DatePicker", datePicker.resolveWidgetVar(context), clientId);
 
         String defaultDate = null;
 
-        if (datepicker.isConversionFailed()) {
-            Class<?> dateType = resolveDateType(context, datepicker);
-            defaultDate = CalendarUtils.getValueAsString(context, datepicker, CalendarUtils.now(uicalendar, dateType));
+        if (datePicker.isConversionFailed()) {
+            Class<?> dateType = resolveDateType(context, datePicker);
+            defaultDate = CalendarUtils.getValueAsString(context, datePicker, CalendarUtils.now(uicalendar, dateType));
         }
         else if (!isValueBlank(value)) {
             defaultDate = value;
         }
 
         wb.attr("defaultDate", defaultDate, null)
-            .attr("inline", datepicker.isInline())
+            .attr("inline", datePicker.isInline())
             .attr("userLocale", locale.toString())
             .attr("dateFormat", CalendarUtils.convertPattern(pattern))
-            .attr("showIcon", datepicker.isShowIcon(), false)
-            .attr("buttonTabindex", datepicker.getButtonTabindex())
-            .attr("focusOnSelect", datepicker.isFocusOnSelect(), false)
-            .attr("disabled", datepicker.isDisabled(), false)
-            .attr("valid", datepicker.isValid(), true)
-            .attr("yearRange", datepicker.getYearRange(), null)
-            .attr("minDate", getMinMaxDate(context, datepicker, datepicker.getMindate(), false), null)
-            .attr("maxDate", getMinMaxDate(context, datepicker, datepicker.getMaxdate(), true), null)
-            .attr("selectionMode", datepicker.getSelectionMode(), null)
-            .attr("showOnFocus", datepicker.isShowOnFocus())
-            .attr("shortYearCutoff", datepicker.getShortYearCutoff(), null)
-            .attr("monthNavigator", datepicker.isMonthNavigator(), false)
-            .attr("yearNavigator", datepicker.isYearNavigator(), false)
-            .attr("showButtonBar", datepicker.isShowButtonBar(), false)
-            .attr("panelStyleClass", datepicker.getPanelStyleClass(), null)
-            .attr("panelStyle", datepicker.getPanelStyle(), null)
-            .attr("keepInvalid", datepicker.isKeepInvalid(), false)
-            .attr("maxDateCount", datepicker.getMaxDateCount(), Integer.MAX_VALUE)
-            .attr("numberOfMonths", datepicker.getNumberOfMonths(), 1)
-            .attr("view", datepicker.getView(), null)
-            .attr("touchUI", datepicker.isTouchUI(), false)
-            .attr("showWeek", datepicker.isShowWeek(), false)
-            .attr("appendTo", SearchExpressionFacade.resolveClientId(context, datepicker, datepicker.getAppendTo(),
+            .attr("showIcon", datePicker.isShowIcon(), false)
+            .attr("buttonTabindex", datePicker.getButtonTabindex())
+            .attr("focusOnSelect", datePicker.isFocusOnSelect(), false)
+            .attr("disabled", datePicker.isDisabled(), false)
+            .attr("valid", datePicker.isValid(), true)
+            .attr("yearRange", datePicker.getYearRange(), null)
+            .attr("minDate", getMinMaxDate(context, datePicker, datePicker.getMindate(), false), null)
+            .attr("maxDate", getMinMaxDate(context, datePicker, datePicker.getMaxdate(), true), null)
+            .attr("selectionMode", datePicker.getSelectionMode(), null)
+            .attr("showOnFocus", datePicker.isShowOnFocus())
+            .attr("shortYearCutoff", datePicker.getShortYearCutoff(), null)
+            .attr("monthNavigator", datePicker.isMonthNavigator(), false)
+            .attr("yearNavigator", datePicker.isYearNavigator(), false)
+            .attr("showButtonBar", datePicker.isShowButtonBar(), false)
+            .attr("panelStyleClass", datePicker.getPanelStyleClass(), null)
+            .attr("panelStyle", datePicker.getPanelStyle(), null)
+            .attr("keepInvalid", datePicker.isKeepInvalid(), false)
+            .attr("maxDateCount", datePicker.getMaxDateCount(), Integer.MAX_VALUE)
+            .attr("numberOfMonths", datePicker.getNumberOfMonths(), 1)
+            .attr("view", datePicker.getView(), null)
+            .attr("touchUI", datePicker.isTouchUI(), false)
+            .attr("showWeek", datePicker.isShowWeek(), false)
+            .attr("appendTo", SearchExpressionFacade.resolveClientId(context, datePicker, datePicker.getAppendTo(),
                             SearchExpressionUtils.SET_RESOLVE_CLIENT_SIDE), null)
-            .attr("icon", datepicker.getTriggerButtonIcon(), null)
-            .attr("rangeSeparator", datepicker.getRangeSeparator(), "-")
-            .attr("timeSeparator", datepicker.getTimeSeparator(), ":")
-            .attr("timeInput", datepicker.isTimeInput())
-            .attr("touchable", ComponentUtils.isTouchable(context, datepicker), true)
-            .attr("lazyModel", datepicker.getModel() instanceof LazyDateMetadataModel, false);
+            .attr("icon", datePicker.getTriggerButtonIcon(), null)
+            .attr("rangeSeparator", datePicker.getRangeSeparator(), "-")
+            .attr("timeSeparator", datePicker.getTimeSeparator(), ":")
+            .attr("timeInput", datePicker.isTimeInput())
+            .attr("touchable", ComponentUtils.isTouchable(context, datePicker), true)
+            .attr("lazyModel", datePicker.getModel() instanceof LazyDateMetadataModel, false);
 
-        List<Integer> disabledDays = datepicker.getDisabledDays();
+        List<Integer> disabledDays = datePicker.getDisabledDays();
         if (disabledDays != null) {
-            CalendarUtils.encodeListValue(context, datepicker, "disabledDays", disabledDays, pattern);
+            CalendarUtils.encodeListValue(context, datePicker, "disabledDays", disabledDays, pattern);
         }
 
-        List<Object> disabledDates = datepicker.getInitialDisabledDates(context);
+        List<Object> disabledDates = datePicker.getInitialDisabledDates(context);
         if (disabledDates != null) {
-            CalendarUtils.encodeListValue(context, datepicker, "disabledDates", disabledDates, pattern);
+            CalendarUtils.encodeListValue(context, datePicker, "disabledDates", disabledDates, pattern);
         }
 
-        String dateTemplate = datepicker.getDateTemplate();
+        String dateTemplate = datePicker.getDateTemplate();
         if (dateTemplate != null) {
             wb.nativeAttr("dateTemplate", dateTemplate);
         }
 
-        String beforeShow = datepicker.getBeforeShow();
+        String beforeShow = datePicker.getBeforeShow();
         if (beforeShow != null) {
             wb.nativeAttr("preShow", beforeShow);
         }
 
-        String onMonthChange = datepicker.getOnMonthChange();
+        String onMonthChange = datePicker.getOnMonthChange();
         if (onMonthChange != null) {
             wb.nativeAttr("onMonthChange", onMonthChange);
         }
 
-        String onYearChange = datepicker.getOnYearChange();
+        String onYearChange = datePicker.getOnYearChange();
         if (onYearChange != null) {
             wb.nativeAttr("onYearChange", onYearChange);
         }
 
-        String weekCalculator = datepicker.getWeekCalculator();
+        String weekCalculator = datePicker.getWeekCalculator();
         if (weekCalculator != null) {
             wb.nativeAttr("weekCalculator", weekCalculator);
         }
 
-        if (datepicker.isShowOtherMonths()) {
-            wb.attr("showOtherMonths", true).attr("selectOtherMonths", datepicker.isSelectOtherMonths());
+        if (datePicker.isShowOtherMonths()) {
+            wb.attr("showOtherMonths", true).attr("selectOtherMonths", datePicker.isSelectOtherMonths());
         }
 
-        if (datepicker.hasTime()) {
-            wb.attr("showTime", datepicker.isShowTime(), false)
-                .attr("hourFormat", datepicker.getHourFormat(), null)
-                .attr("timeOnly", datepicker.isTimeOnly(), false)
-                .attr("showSeconds", datepicker.isShowSeconds(), false)
-                .attr("stepHour", datepicker.getStepHour(), 1)
-                .attr("stepMinute", datepicker.getStepMinute(), 1)
-                .attr("stepSecond", datepicker.getStepSecond(), 1)
-                .attr("hideOnDateTimeSelect", datepicker.isHideOnDateTimeSelect(), false);
+        if (datePicker.hasTime()) {
+            wb.attr("showTime", datePicker.isShowTime(), false)
+                .attr("hourFormat", datePicker.getHourFormat(), null)
+                .attr("timeOnly", datePicker.isTimeOnly(), false)
+                .attr("showSeconds", datePicker.isShowSeconds(), false)
+                .attr("stepHour", datePicker.getStepHour(), 1)
+                .attr("stepMinute", datePicker.getStepMinute(), 1)
+                .attr("stepSecond", datePicker.getStepSecond(), 1)
+                .attr("hideOnDateTimeSelect", datePicker.isHideOnDateTimeSelect(), false);
         }
 
-        String mask = datepicker.getMask();
+        String mask = datePicker.getMask();
         if (mask != null && !mask.equals("false")) {
-            String patternTemplate = datepicker.calculatePattern();
-            String maskTemplate = (mask.equals("true")) ? datepicker.convertPattern(patternTemplate) : mask;
+            String patternTemplate = datePicker.calculatePattern();
+            String maskTemplate = (mask.equals("true")) ? datePicker.convertPattern(patternTemplate) : mask;
             wb.attr("mask", maskTemplate)
-                .attr("maskSlotChar", datepicker.getMaskSlotChar(), "_")
-                .attr("maskAutoClear", datepicker.isMaskAutoClear(), true);
+                .attr("maskSlotChar", datePicker.getMaskSlotChar(), "_")
+                .attr("maskAutoClear", datePicker.isMaskAutoClear(), true);
         }
 
-        encodeClientBehaviors(context, datepicker);
+        encodeClientBehaviors(context, datePicker);
 
         wb.finish();
     }
 
     @Override
     public Object getConvertedValue(FacesContext context, UIComponent component, Object value) throws ConverterException {
-        DatePicker datepicker = (DatePicker) component;
+        DatePicker datePicker = (DatePicker) component;
         String submittedValue = (String) value;
 
         if (isValueBlank(submittedValue)) {
             return null;
         }
 
-        String selectionMode = datepicker.getSelectionMode();
+        String selectionMode = datePicker.getSelectionMode();
         switch (selectionMode) {
             case "multiple": {
                 String[] parts = submittedValue.split(",");
@@ -287,7 +287,7 @@ public class DatePickerRenderer extends BaseCalendarRenderer {
                 return multi;
             }
             case "range": {
-                String[] parts = submittedValue.split(datepicker.getRangeSeparator());
+                String[] parts = submittedValue.split(datePicker.getRangeSeparator());
                 List<Object> range = new ArrayList<>();
                 if (parts.length == 2) {
                     for (String part : parts) {
@@ -302,12 +302,12 @@ public class DatePickerRenderer extends BaseCalendarRenderer {
         }
     }
 
-    private String getMinMaxDate(FacesContext context, DatePicker datepicker, Object value, boolean max) {
-        if (value instanceof LocalDate && datepicker.isShowTime()) {
+    private String getMinMaxDate(FacesContext context, DatePicker datePicker, Object value, boolean max) {
+        if (value instanceof LocalDate && datePicker.isShowTime()) {
             LocalDate date = (LocalDate) value;
             value = date.atTime(max ? LocalTime.MAX : LocalTime.MIN);
         }
-        return CalendarUtils.getValueAsString(context, datepicker, value);
+        return CalendarUtils.getValueAsString(context, datePicker, value);
     }
 }
 

